zar index does not handle more than 5 "levels"

I was trying to achieve zar index parity with our older, closed-source indexing, so I ran `zar index :bstring` on a large data set. After a few minutes, there was a panic with the text:
```
file:///mnt/brim-workspace/zar/20180324/1521931333.963907.zng: creating index file:///mnt/brim-workspace/zar/20180324/1521931333.963907.zng.zar/zdx-type-bstring
panic: something wrong

goroutine 1 [running]:
github.com/brimsec/zq/zdx.newWriter(0xc012f22150, 0xc00fcb1440, 0x4, 0x0, 0x0, 0x0, 0xc00fcb1447, 0x0, 0xc00fcb1447, 0x4b, ...)
        /home/mikeb/zq/zdx/writer.go:73 +0x714
```
I checked the code, and this isn't a bug per se, but a [level assertion](https://github.com/brimsec/zq/blob/588c08edce5101cb5e448732e41aaedd27a16cce/zdx/writer.go#L73). More than 5 "levels" aren't allowed.
```go
func newWriter(zctx *resolver.Context, path iosrc.URI, keyFields []string, framesize, level int, hdr *zng.Record) (*Writer, error) {
	if level > 5 {
		panic("something wrong")
	}
```
Given the following, I assume this is related to the number of distinct microindices per chunk.
```
-rw-rw-r-- 1 mikeb mikeb 654K Jul  9 22:36 /mnt/brim-workspace/zar/20180324/1521931333.963907.zng.zar/zdx-type-bstring.1.zng
-rw-rw-r-- 1 mikeb mikeb 519K Jul  9 22:36 /mnt/brim-workspace/zar/20180324/1521931333.963907.zng.zar/zdx-type-bstring.2.zng
-rw-rw-r-- 1 mikeb mikeb 390K Jul  9 22:36 /mnt/brim-workspace/zar/20180324/1521931333.963907.zng.zar/zdx-type-bstring.3.zng
-rw-rw-r-- 1 mikeb mikeb 260K Jul  9 22:36 /mnt/brim-workspace/zar/20180324/1521931333.963907.zng.zar/zdx-type-bstring.4.zng
-rw-rw-r-- 1 mikeb mikeb 130K Jul  9 22:36 /mnt/brim-workspace/zar/20180324/1521931333.963907.zng.zar/zdx-type-bstring.5.zng
-rw-rw-r-- 1 mikeb mikeb 805K Jul  9 22:36 /mnt/brim-workspace/zar/20180324/1521931333.963907.zng.zar/zdx-type-bstring.zng
```
If this is just a bumper to prevent an early bug that didn't imagine this meany zdx's per chunk, maybe it can just go away.

If this is a bumper that's intentional, guidance or other defaults ought to be adjusted for how to overcome the barrier.

--- a/tests/suite/zdx/maxlevels.yaml
+++ b/tests/suite/zdx/maxlevels.yaml
@@ -1,0 +1,10 @@
+script: |
+  zdx create -o index.zng -k s -f 20 babble.tzng
+
+inputs:
+  - name: babble.tzng
+
+outputs:
+  - name: stderr
+    regexp: |
+      .*too many levels.*

--- a/zdx/writer.go
+++ b/zdx/writer.go
@@ -42,6 +42,7 @@ type Writer struct {
 	keyType     *zng.TypeRecord
 	iow         io.WriteCloser
 	childField  string
+	nlevel      int
 }
 
 type indexWriter struct {
@@ -185,6 +186,10 @@ func (w *Writer) finalize() error {
 }
 
 func newIndexWriter(base *Writer, w io.WriteCloser, name string) (*indexWriter, error) {
+	base.nlevel++
+	if base.nlevel >= MaxLevels {
+		return nil, ErrTooManyLevels
+	}
 	writer := bufwriter.New(w)
 	return &indexWriter{
 		base:     base,

--- a/zdx/zdx.go
+++ b/zdx/zdx.go
@@ -19,6 +19,8 @@ import (
 	"errors"
 )
 
+const MaxLevels = 20
+
 var (
-	ErrCorruptFile = errors.New("corrupt zdx file")
+	ErrTooManyLevels = errors.New("microindex has too many levels (a larger frame threshold is needed)")
 )

